lib/models/command.js

```diff
@@ -288,7 +288,11 @@ let Command = CoreObject.extend({
 
       if (this.works === 'insideProject') {
         if (!this.project.hasDependencies()) {
-          throw new SilentError('node_modules appears empty, you may need to run `npm install`');
+          let installInstuction = '`npm install`';
+          if (this.project.hasYarnLockFile()) {
```

Comment:
I'd prefer to check it here directly, and not add another method to `project`. 

If however, we do keep the project method around, we should update other locations that search for `yarn.lock` to use it ([example here](https://github.com/ember-cli/ember-cli/blob/bd2822868e3b9bf11cfa9f28b91cb4d021c6680b/lib/commands/init.js#L64)).